### ISSUE TITLE
Relative imports paths when generating multiple models

### ISSUE THREAD

**Issue by lord-haffi** [koxudaxi/datamodel-code-generator#1683]
When generating schemas with a directory of JSON-Schemas as `--input` the relative import paths will be problematic. I had a complex use-case where pydantic threw an error when attempting to import such a generated model. But maybe I'm missing something.

So, e.g. if you have a structure like
```
- src
    - pkg1
        - a.py
    - pkg2
        - b.py
```
a module would look like this:
```python
from ..pkg2 import b
from pydantic import BaseModel

class A(BaseModel):
    bb: b.B | None = None
```

Which is fine in principle. But when attempting to import e.g. class `A` which references to an enum, pydantic throws an error similar to `B has no attribute B`. So this might be more an issue of pydantic. I would suggest to change the relative import to import the model directly (or at least have a flag for this) anyway.

```python
from ..pkg2.b import B
from pydantic import BaseModel

class A(BaseModel):
    bb: B | None = None
```

**Comment by koxudaxi** [comment]
@lord-haffi 
>  So this might be more an issue of pydantic. I would suggest to change the relative import to import the model directly (or at least have a flag for this) anyway.


I'm sorry. I need help understanding your faced problem.
Can you explain in more detail?

Can you resolve the Relative imports problem for your suggestion?

I tried your example. I got the relative imports errors.
```python
>>> from pkg1.a import A
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/koudai/PycharmProjects/pythonProject/output/pkg1/a.py", line 11, in <module>
    from ..pkg2 import b
ImportError: attempted relative import beyond top-level package
```

```python
>>> from pkg1.a import A
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/koudai/PycharmProjects/pythonProject/output/pkg1/a.py", line 11, in <module>
    from ..pkg2.b import B
ImportError: attempted relative import beyond top-level package
>>> 
```

Do we need `__init__.py` in `src` directory to detect src as a package?

**Comment by lord-haffi** [comment]
Sorry for my late response. I have not so much time and as I commented in #1684, I monkey-patched the function for me, so it doesn't have prio for me.

I don't know what you did there. As for example [this StackOverflow answer](https://stackoverflow.com/a/47030746/21303427) explains, these kind of import errors usually appear if you don't "call" the package correctly. Judging by the paths of error messages you posted, I think that output is missing a `__init__.py` file if that is what you meant. But I don't know how you did this since in my experiments it always generated such a `__init__.py` file.

**Comment by lord-haffi** [comment]
Fixed by #1983
